crates/release_plz_core/src/repo_versions.rs

```diff
@@ -0,0 +1,76 @@
+use cargo_metadata::semver::Version;
+use git_cmd::Repo;
+use regex::Regex;
+use std::collections::BTreeMap;
+
+lazy_static::lazy_static! {
+    // match PR/issue numbers, e.g. `#123`
+    static ref VERSION_TAG_RE: Regex = Regex::new("(?:(.+)-)?v(.+)").unwrap();
+}
+
+pub enum RepoVersions {
+    /// Most recent version for the repository's single public package.
+    Single(Version),
+    /// Most recent version for each of the repository's public packages
+    ByPackage(BTreeMap<String, Version>),
+}
+
+impl RepoVersions {
+    pub fn get_package_version(&self, package_name: &str) -> Option<&Version> {
+        match self {
+            Self::Single(version) => Some(version),
+            Self::ByPackage(versions) => versions.get(package_name),
+        }
+    }
+}
+
+/// Gets sorted tags from `Repo` and returns:
+/// * `None` if there are no version tags
+/// * `RepoVersions::Single` if all verison tags are of the form 'v{version}' and
+///   `contains_multiple_pub_packages` is `false`
+/// * `RepoVersions::ByPackage` if all verison tags are of the form '{package}-v{version}' and
+///   `contains_multiple_pub_packages` is `true`
+/// * Panics otherwise.
```

Comment:
can you return anyhow::Result instead of panicking?

Reply:
Done!